### PR TITLE
fix(Core/Spells): Script Leggings of Beast Mastery.

### PR DESCRIPTION
**Pull request by avarishd**
## Changes Proposed:
This PR proposes changes to:
-  [] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15895
- Closes https://github.com/chromiecraft/chromiecraft/issues/5363

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

Make a hunter
.npc add temp 33286
.pet create
.setskill 413 400
.additem 31328

Equip some more items with different stats
Calculate pet bonuses - 
Player armor  * 0.35 + 490
Player stamina *  0.45 + 52
Player rangedAP * 0.22 + 70 

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 1]
```diff
@@ -165,6 +166,10 @@ class spell_hun_generic_scaling : public AuraScript
             SpellSchoolMask schoolMask = SpellSchoolMask(aurEff->GetSpellInfo()->Effects[aurEff->GetEffIndex()].MiscValue);
             int32 modifier = schoolMask == SPELL_SCHOOL_MASK_NORMAL ? 35 : 40;
             amount = CalculatePct(std::max<int32>(0, owner->GetResistance(schoolMask)), modifier);
+            if (owner->HasAura(SPELL_HUNTER_PET_LEGGINGS_OF_BEAST_MASTERY) && schoolMask == SPELL_SCHOOL_MASK_NORMAL)
+            {
+                amount += 490;
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_hunter.cpp:171]: famous question, how did we come up with this

Comment (avarishd) [reply src/server/scripts/Spells/spell_hunter.cpp:171]: https://www.wowhead.com/wotlk/item=31328/leggings-of-beast-mastery

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_hunter.cpp:171]: Love you

Comment (avarishd) [reply src/server/scripts/Spells/spell_hunter.cpp:171]: As much as I want to conjure random numbers, you won't let me 😠

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_hunter.cpp:171]: But you love me anyway, dont you :D

Comment (avarishd) [reply src/server/scripts/Spells/spell_hunter.cpp:171]: https://www.youtube.com/watch?v=9wtvXoXh0VU

Comment (Nyeriah) [reply src/server/scripts/Spells/spell_hunter.cpp:171]: Is it not part of the spell attribute? Should use spell/aura attribute instead of hardcoded number

Comment (avarishd) [reply src/server/scripts/Spells/spell_hunter.cpp:171]: I could, however do it for https://github.com/azerothcore/azerothcore-wotlk/pull/17427

**Review by Nefertumm — APPROVED** [review]

**Comment by Nefertumm** [comment]
Confirmed the spell doesn't have misc values or base points to get the attributes from spellinfo.

**Pull request merged**